### DISCUSSION TITLE
Hello, Reading data from Hycom.

### DISCUSSION THREAD

**Discussion by manueliturra** [Parcels-code/Parcels#1533] [General]
Hello, im working with  Inertial Oscilaltions using parcels. I always use parcels with data at one specific depth and specific time windows, now I download Hycom data, but the variables are u0 = 32,32,40,200 = x,y,depth,time. The HYCOM FILE have 20 depth in a 4d matrix?, I’m new with python and I dont find any example in the tutorial section. Basically, I want to chose an specific time window, like from day 6(6 JAN) to day 10(10 JAN), and only selec one depth.

In the following section, I dont know where to only select one depth like u0(:,:,1,:) for the fildset reading.
I will apreciate any help, i also will apreciate if i can simulate inertial oscilations with ocean parcels, thank you.

# Names of files containing the data
file = "cmems_mod_glo_phy_my_0.083deg_P1D-m_uo-vo_75.25W-71.83W_37.92S-34.17S_0.49m_2005-01-01-2005-01-30.nc"
filenames = {
    'U': file,
    'V': file
}

# Names of the velocity component data in the file
variables = {
    'U': 'uo',
    'V': 'vo'
}

# Names of the dimensions in the file
dimensions = {
    'lat': 'latitude',
    'lon': 'longitude',
    'time': 'time'
}

fieldset = FieldSet.from_netcdf(filenames, variables, dimensions,vmin=-2000,allow_time_extrapolation=True)

**Comment by michaeldenes** [comment]
Hi @manueliturra, one way you can approach this is using the `indices` argument in `FieldSet.from_netcdf`. This is usually used to select a particular latitude/longitude range or depth range, but I can't see why it couldn't be used for time!

E.g. `indices = {'time': range(0,10), 'depth': range(15,16)}` will use the first 10 time units of data (you will have to manually figure out when 6 Jan - 10 Jan is), and the 15th depth index. You then include this in your `from_netcdf` call, e.g. `fieldset = FieldSet.from_netcdf(filenames, variables, dimensions, indices=indices)`.

Let us know if that solves the problem for you, cheers!

**Reply by michaeldenes** [reply]
Hi @manueliturra, apologies for the delay in responding. An alternative is to first open the dataset using xarray, e.g. `ds = xr.open_mfdataset(filenames)`, then subsettings your xarray dataset `ds`, and using the parcels `from_xarray()` method. This would be the simplest option, adding only two lines of code to your notebook! Let me know how that goes, cheers!
